Public/Src/Engine/Scheduler/Distribution/Worker.cs

```diff
@@ -421,27 +421,54 @@ internal bool TryAcquire(RunnablePip runnablePip, out WorkerResource? limitingRe
                 loadFactor = 1;
             }
 
-            if (AcquiredProcessSlots >= (EffectiveTotalProcessSlots * loadFactor))
+            var processRunnablePip = runnablePip as ProcessRunnablePip;

```

Comment:
> as  [](start = 48, length = 4)

It seems the code assumes it can only happen with a `ProcessRunnalbePip`, so please use a cast, so it fails with a CastException rather than a NullRefException later.

Reply:
I can add a cast instead. 

There was actually already an implicit assumption that only processRunnablePips would get here due to the processRunnable != null check on what used to be line 432 and the Contract.Ensures at the top of the function.

---
In reply to: [271091922](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271091922) [](ancestors = 271091922)

Reply:
In that case just add a `Contract.Requires` to this method

Reply:
Yeah I actually added a Contract.Assert down below, I just forgot about it.

---
In reply to: [271104950](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271104950) [](ancestors = 271104950)

Reply:
Added a contract requirement instead.

---
In reply to: [271093475](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271093475) [](ancestors = 271093475,271091922)